src/reconstruction/modelGeneration/massBalance/computeMW.m

```diff
@@ -31,75 +38,44 @@
 
 metIDs = reshape(metIDs, length(metIDs),1);
 
-MW = zeros(size(metIDs));
-for n = 1:length(metIDs)
-    i = metIDs(n);
-    formula = model.metFormulas(i);
-    [compounds, tok] = regexp(formula, '([A-Z][a-z]*)(\d*)', 'match', 'tokens');
-    tok = tok{1,1};
-    for j = 1:length(tok) % go through each token.
-        t = tok{1,j};
-        comp = t{1,1};
-        q = str2num(t{1,2});
-        if (isempty(q))
-            q = 1;
-        end
-        mwt = 0;
-        switch comp
-            case 'H'
-                mwt = 1;
-            case 'C'
-                mwt = 12;
-            case 'N'
-                mwt = 14;
-            case 'O'
-                mwt = 16;
-            case 'Na'
-                mwt = 23;
-            case 'Mg'
-                mwt = 24;
-            case 'P'
-                mwt = 31;
-            case 'S'
-                mwt = 32;
-            case 'Cl'
-                mwt = 35;
-            case 'K'
-                mwt = 39;
-            case 'Ca'
-                mwt = 40;
-            case 'Mn'
-                mwt = 55;
-            case 'Fe'
-                mwt = 56;
-            case 'Ni'
-                mwt = 58;
-            case 'Co'
-                mwt = 59;
-            case 'Cu'
-                mwt = 63;
-            case 'Zn'
-                mwt = 65;
-            case 'As'
-                mwt = 75;
-            case 'Se'
-                mwt = 80;
-            case 'Ag'
-                mwt = 107;
-            case 'Cd'
-                mwt = 114;
-            case 'W'
-                mwt = 184;
-            case 'Hg'
-                mwt = 202;
-            otherwise
-                if warnings
-                    display('Warning');
-                    display(formula)
-                    display(comp);
-                end
+% molecular weight of elements in a structure
+elementMwStruct = struct('H', 1, 'C', 12, 'N', 14, 'O', 16, 'Na', 23, 'Mg', 24, 'P', 31, ...
+    'S', 32, 'Cl', 35, 'K', 39, 'Ca', 40, 'Mn', 55, 'Fe', 56, 'Ni', 58, 'Co', 59, ...
+    'Cu', 63, 'Zn', 65, 'As', 75, 'Se', 80, 'Ag', 107, 'Cd', 114, 'W', 184, 'Hg', 202);
+elementNames = fieldnames(elementMwStruct);  % elements' names

```

Comment:
Would it make sense to:
A: create a script that extracts this information from a database and writes it to a csv and 
B: load it from that csv? 
Along those lines (apart from probably otherwise failing a test), is there a reason that the weights are given as integer values instead of the isotope averages comonly available in databases?

Reply:
I just tried to stick to the original output. Sure it is better to do what you said, as discussed above.